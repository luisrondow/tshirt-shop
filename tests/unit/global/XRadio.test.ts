import { describe, it, expect } from 'vitest'
import { mount } from '@vue/test-utils'

import { XRadio } from '#components'

describe('XRadio component', () => {
  it('should render correctly', () => {
    const wrapper = mount(XRadio)

    expect(wrapper.vm).toBeTruthy()
    expect(wrapper.element).toMatchSnapshot()
  })

  it('should render correctly with props', () => {
    const wrapper = mount(XRadio, {
      props: {
        value: 'test',
        label: 'Test',
        name: 'test',
        id: 'test-1',
        checked: true,
      },
    })

    expect(wrapper.find('input').element.value).toBe('test')
    expect(wrapper.find('label').text()).toBe('Test')
    expect(wrapper.find('input').attributes('name')).toBe('test')
    expect(wrapper.find('input').attributes('id')).toBe('test-1')
    expect(wrapper.find('input').element.checked).toBeTruthy()
  })

  it('should check radio works correctly', async () => {
    const wrapper = mount(XRadio, {
      props: {
        value: 'test',
        label: 'Test',
        name: 'test',
        id: 'test-1',
        checked: false,
      },
    })

    const input = wrapper.find('input')

    input.trigger('click')
    input.trigger('change')

    expect(wrapper.find('input').element.checked).toBeTruthy()
    expect(wrapper.emitted('update:value')).toBeTruthy()
    expect(wrapper.emitted('update:value')?.[0]).toEqual(['test'])
  })
})
