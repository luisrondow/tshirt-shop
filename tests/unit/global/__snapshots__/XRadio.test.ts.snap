// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`XRadio component > should render correctly 1`] = `
<div
  class="mb-4 flex items-center"
>
  <input
    class="h-4 w-4 border-gray-300 bg-gray-100 text-black"
    id=""
    name=""
    type="radio"
    value=""
  />
  <label
    class="ms-2 text-lg font-medium text-gray-900"
    for=""
  />
</div>
`;
